src/test/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterControllerTest.java

```diff
@@ -122,18 +150,69 @@ public void should_return_json_processing_exception_when_service_fails_to_serial
 
     }
 
-
     @Test
     public void should_return_400_client_error_when_invalid_letter_is_sent() throws Exception {
         sendLetter("").andExpect(status().is4xxClientError());
 
         verify(letterService, never()).send(any(Letter.class), anyString());
     }
 
+    @Test
+    public void should_return_400_client_error_when_letter_is_sent_without_documents() throws Exception {
+        sendLetter(LETTER_WITHOUT_DOCUMENTS_JSON)
+            .andExpect(status().is4xxClientError())
+            .andExpect(content()
+                .json("{\"errors\":[{\"field_name\":\"documents\",\"message\":\"size must be between 1 and 10\"}]}"));
+
+        verify(letterService, never()).send(any(Letter.class), anyString());
+    }
+
+    @Test
+    public void should_return_400_client_error_when_letter_is_sent_without_type() throws Exception {
+        sendLetter(LETTER_WITHOUT_TYPE_JSON)
+            .andExpect(status().is4xxClientError())
+            .andExpect(content()
+                .json("{\"errors\":[{\"field_name\":\"type\",\"message\":\"may not be empty\"}]}"));
+
+        verify(letterService, never()).send(any(Letter.class), anyString());
+    }
+
+    @Test
+    public void should_return_400_client_error_when_letter_is_sent_without_template_in_document() throws Exception {
+        sendLetter(LETTER_WITHOUT_DOC_TEMPLATE_JSON)
+            .andExpect(status().is4xxClientError())
+            .andExpect(content()
+                .json("{\"errors\":[{\"field_name\":\"documents[0].template\",\"message\":\"may not be empty\"}]}"));
+
+        verify(letterService, never()).send(any(Letter.class), anyString());
+    }
+
+    @Test
+    public void should_return_400_client_error_when_letter_is_sent_without_template_values_in_document()
+        throws Exception {
+        sendLetter(LETTER_WITHOUT_DOC_TEMPLATE_VALUES_JSON)
+            .andExpect(status().is4xxClientError())
+            .andExpect(content()
+                .json("{\"errors\":[{\"field_name\":\"documents[0].values\",\"message\":\"may not be empty\"}]}"));
+
+        verify(letterService, never()).send(any(Letter.class), anyString());
+    }
+
+    @Test
+    public void should_return_400_client_error_when_letter_is_with_more_than_10_documents()
+        throws Exception {
+        sendLetter(LETTER_WITH_MORE_THAN_TEN_DOCS_JSON)
+            .andExpect(status().is4xxClientError())
+            .andExpect(content()
+                .json("{\"errors\":[{\"field_name\":\"documents\",\"message\":\"size must be between 1 and 10\"}]}"));
+
+        verify(letterService, never()).send(any(Letter.class), anyString());
+    }
+
```

Comment:
can use `jsonPath` to validate response but i guess the point is to validate the whole thing so not bothered as at the time

Reply:
nevertheless - extensive unit coverage 👏